db/migrate/20231024193006_add_route_param_to_domain_admin_action_logs.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class AddRouteParamToDomainAdminActionLogs < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def up
+    Admin::ActionLog.unscoped.where(target_type: %w(DomainBlock DomainAllow Instance UnavailableDomain)).find_each do |action_log|
+      # We don't actually need to look at the target, as the human identifier is
+      # the value we want for the route_param
+      action_log.update(route_param: action_log.human_identifier)
+    end
```

Comment:
Is there a better way to write this migration? Should it be a post deployment migration?

Reply:
In cases like these, I like to drop down to raw SQL. Something like

```ruby
execute("UPDATE admin_action_logs SET route_param = human_identifier WHERE target_type IN ('DomainBlock', 'DomainAllow', 'Instance', 'UnavailableDomain')")
```

I think you might be able to construct the same using ActiveRecord using something like this:

```ruby
Admin::ActionLog.unscoped.where(target_type: %w(DomainBlock DomainAllow Instance UnavailableDomain)).update_all(route_param: Arel.sql('human_identifier'))
```

> Should it be a post deployment migration?

Great questions. I have to admit I have not fully internalized when to use what yet. But I think yes. I believe there are two important questions to answer: Will the new code once deployed work without this? And will this take a long time?

The answers here are "somewhat" and "very probably". So while the code would not show all records until the migration is finished it would not raise an exception either. And with a lot of action logs the migration might take a while.

Reply:
Have removed this migration, as discussed.